firmware/src/tasks/task_buzzer.cpp

```diff
@@ -0,0 +1,51 @@
+/*
+ * CATS Flight Software
+ * Copyright (C) 2023 Control and Telemetry Systems
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <https://www.gnu.org/licenses/>.
+ */
+
+#include "tasks/task_buzzer.h"
+#include "config/cats_config.h"
+
+namespace task {
+
+void Buzzer::Beep(BeepCode code) const { osEventFlagsSet(m_buzzer_event_id, static_cast<uint32_t>(code)); }
+
+[[noreturn]] void Buzzer::Run() noexcept {
+  while (true) {
+    /* Check Flight Phases */
+    auto id = osEventFlagsWait(m_buzzer_event_id, 0xFFU, osFlagsWaitAny, osWaitForever);
```

Comment:
What happens if during the beeping (so while we are in the for loop on line 33) more buzzer events are set? Are things added to the event overwriting the old event or is it or'd?